doc/contribute.rst

```diff
@@ -0,0 +1,329 @@
+"enhancement". If you already have an implementation of the idea, open a
+`pull request <https://github.com/NeuralEnsemble/elephant/pulls>`_.
+
+.. _set_up_an_environment:
+
+*************************
+Setting up an environment
+*************************
+
+To make any changes to Elephant code, you must set up a Python environment on
+your local machine.
+
+1. Follow the instructions in :ref:`prerequisites` to setup a clean conda
+   environment. To be safe, run::
+
+    $ pip uninstall elephant
+
+   to uninstall ``elephant`` in case you've installed it previously as a pip
+   package.
+
+2. Fork `Elephant <https://github.com/NeuralEnsemble/elephant>`_ as described
+   in `Fork a repo <https://help.github.com/en/github/getting-started-with-github/fork-a-repo>`_.
+   Download Elephant source code from your forked repo::
+
+    $ git clone git://github.com/<your-github-profile>/elephant.git
+    $ cd elephant
+
+3. Install the requirements (either via pip or conda):
+
+.. tabs::
+
+    .. tab:: pip
+
+        .. code-block:: sh
+
+            pip install -r requirements/requirements.txt
+            pip install -r requirements/requirements-extras.txt  # optional
+            pip install -r requirements/requirements-tests.txt
+
+    .. tab:: conda
+
+        .. code-block:: sh
+
+            conda env create -f requirements/environment.yml
+            conda activate elephant
+            pip install -r requirements/requirements-tests.txt
+
+
+4. Before you make any changes, run the test suite to make sure all the tests
+   pass on your system::
+
+    $ nosetests .
+
+   You can specify a particular module to test, for example
+   ``test_statistics.py``::
+
+    $ nosetests elephant/test/test_statistics.py
+
+   At the end, if you see "OK", then all the tests passed (or were skipped
+   because certain dependencies are not installed), otherwise it will report
+   on tests that failed or produced errors.
+
+
+************************
+Adding new functionality
+************************
+
+After you've set up a development environment, implement a functionality you
+want to add in Elephant. This includes (either of):
+
+   * fixing a bug;
+   * improving the documentation;
+   * adding a new functionality.
+
+Writing the code
+----------------
+
+Imagine that you want to add in existing module ``elephant/statistics.py`` some
+statistics calculation of a list of spike trains that returns an analog signal.
+Let's call it ``statistics_x``.
+
+Elephant relies on Neo structures that use Quantities extensively, allowing
+the users not to care about the different units (seconds and milliseconds)
+of input objects. Typically, to avoid computationally expensive quantities
+rescaling operation on large input arrays, we check that the main data objects
+- spike trains or analog signals - share the same units and rescale additional
+parameters (t_start, t_stop, bin_size, etc.) to the units of input objects.
+
```

Comment:
I didn't understand your message.

Reply:
I meant that currently, the step-by-step section on contributing code is separated from the section on how to document. I thought maybe it would be good refer to the documentation section below, to avoid that people miss reading it. Perhaps a sentence like "For guidelines on code documentation, please see the Documentation Guide below.", or even more explicitly for code documentation, "For guidelines on code documentation, please see the Code Style Guide with Examples."